iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/functional_interface.inl

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_CXX_FUNCTIONAL_INTERFACE_INL
+#define IOX_HOOFS_CXX_FUNCTIONAL_INTERFACE_INL
+
+namespace iox
+{
+namespace cxx
+{
+namespace internal
+{
+template <typename Derived>
+inline void Expect<Derived>::expect(const char* const msg) const noexcept
+{
+    if (!(*static_cast<const Derived*>(this)))
+    {
+        std::cout << msg << std::endl;
+        Ensures(false);
+    }
+}
+
+template <typename Derived, typename ValueType>
+inline ValueType& ExpectWithValue<Derived, ValueType>::expect(const char* const msg) & noexcept
+{
+    Derived* derivedThis = static_cast<Derived*>(this);
+
+    if (!(*derivedThis))
+    {
+        std::cout << msg << std::endl;
+        Ensures(false);
+    }
+
+    return derivedThis->value();
+}
+
+template <typename Derived, typename ValueType>
+inline const ValueType& ExpectWithValue<Derived, ValueType>::expect(const char* const msg) const& noexcept
+{
+    return const_cast<const ValueType&>(const_cast<ExpectWithValue<Derived, ValueType>*>(this)->expect(msg));
```

Comment:
What do you think about having a `using Self = ExpectWithValue<Derived, ValueType>` alias in the class definition and use it here?

Reply:
I added the Self alias in the method scope here and in some other places.